feat(docs): add markdown based documentation (#28)
accuknox/report-action: README.md

> # CI/CD Security by AccuKnox

Reviewer: # Monitor & Audit CI/CD pipelines

Author: Addressed.

![](./assets/ak-logo-light-back.png)

## Table of Contents
- [Introduction](#introduction)
- [KubeArmor](#kubearmor)
- [Features](#features)
- [Usage](#usage)
- [Inputs](#inputs)
- [Examples](#examples)
- [Troubleshooting](#troubleshooting)
- [Contributing](#contributing)

## Introduction
The CI/CD Scan by AccuKnox is a powerful tool designed to enhance the security of 
your continuous integration and deployment pipelines. By providing comprehensive 
assessment, monitoring, and protection capabilities, it enables you to:

> - Detect and prevent potential supply chain attacks in CI/CD environments
> - Gain real-time visibility into your pipeline's security posture
> - Streamline the integration of security practices into your DevOps workflow

Reviewer: ```suggestion
- Detect and prevent potential supply chain attacks in CI/CD environments
- Gain real-time visibility into your pipeline's security posture
- Streamline the integration of security practices into your DevOps workflow

Ensure application best practices by:
* Applying app hardening policies and checking whether it deviates during GH workflow execution.
* Identifying if there are any unknown processes spawning during CI/CD workflow execution.
* Identifying if any unwanted network connections are started in the pipeline.
```

Author: Addressed.

In today's fast-paced software development landscape, where operational efficiency 
is paramount, the CI/CD Scan by AccuKnox empowers DevSecOps teams to deploy with confidence. 

## KubeArmor
KubeArmor is a CNCF project that powers this action to scan the events taking place
in the CI/CD environment such as GitHub runner. KubeArmor uses eBPF monitors to 
watch over the events taking place at system level and uses LSMs (Linux security modules)
for enforcing the security policies either in block or audit mode. 

To learn more about KubeArmor please visit, https://kubearmor.io/ 

> We install KubeArmor in systemd mode in the GitHub runner and that lets us watch over

Reviewer: ```suggestion
This Action installs KubeArmor in systemd mode in the GitHub runner and watches over
```

Author: Addressed.

![MITRE ATT&CK](https://avatars.githubusercontent.com/u/43272913?s=200&v=4)|![NIST](https://avatars.githubusercontent.com/u/5330158?s=200&v=4)|![CIS](https://avatars.githubusercontent.com/u/8919353?s=200&v=4)
---------------------------------------------------------------------------|------------------------------------------------------------------|----------------------------------------------------------------
MITRE ATT&CK | NIST | CIS

All of our security policy templates are open sourced and can be found here: [Policy Templates](https://github.com/kubearmor/policy-templates)

The alerts generated after the scan of your CI/CD pipeline is completed looks like this
![](./assets/alert_example_1.png)

It clearly tells you about the policy itself, the command which caused the generation of alert
and the message itself, which makes it easy for you to understand the nature of alert and
a possible issue in the CI/CD pipeline itself.

### 2) Define your own security policies
Other than providing the baseline as an initial security posture, you can also write your
own security policies and have a more granular control on the CI/CD pipeline. 

> Here is an example of a security policy

Reviewer: Remove the below YAML policy and provide a realistic policy from existing template

Author: Added a more realistic policy.

    - path: /usr/bin/sleep # try sleep 1
  action:
    Block
```

Please make sure that in `kind` field you set `KubeArmorHostPolicy`. 
The above policy will block the `sleep` call. To read more about how KubeArmorHost 
policies are written and designed please take a look at this: [KubeArmor policy spec for nodes/VMs](https://docs.kubearmor.io/kubearmor/documentation/host_security_policy_specification)

### 3) Process tree and behaviour
We let you see complete process tree that includes a hierarchical view of the 
processes taking place in the pipeline, we do not exclude a single process from 
the view, making sure its easy for you to do a system forensics based analysis.

Here is an example of it:
![](./assets/pstree_example_1.png)

As you can see, you can clearly observe the parent-child relationship between
processes easily and observe the overall system behaviour.

### 4) Network events
You can also see all the network based events taking place in the system, this 
is very important to identify any egress or ingress network events with the network
protocol, process making the network call and the name of the process itself. 

For example:
![](./network_example_1.png)

> <em>We will also support standard network policies in the coming realeases.</em>

Reviewer: ```suggestion
<em>Support for network policies in the coming releases.</em>
```

Author: Addressed.

## Usage
The usage is as simple as <em>Plug-and-Play</em>, you only have to include the following 
lines in your GitHub workflow and you are all set to go. 

```yaml

> - name: AccuKnox CI/CD scan

Reviewer: ```suggestion
- name: AccuKnox CI/CD Monitor
```

Author: Addressed.
